pages/docs/reference/whatsnew1420.md

```diff
@@ -0,0 +1,348 @@
+project published with earlier versions.
+
+Projects and libraries without the hierarchical project structure remain compatible.
+
+[**Back to top**](#)
+
+## CocoaPods support
+
+### Improved task execution
+
+In this release, we significantly improved task execution flow. For example, if you add a new CocoaPods dependency,
+existing dependencies are not rebuilt. Adding an extra target also doesn't affect rebuilding dependencies for existing ones.
+
+### Extended DSL
+
+In 1.4.20, we extend the DSL of adding [CocoaPods](native/cocoapods.html) dependencies to your Kotlin project.
+
+In addition to local Pods and Pods from the CocoaPods repository, you can add dependencies on the following types of libraries:
+* A library from a custom spec repository.
+* A remote library from a Git repository.
+* A library from an archive (also available by arbitrary HTTP address).
+* A static library.
+* A library with custom cinterop options.
+
+Learn more about adding CocoaPods dependencies in Kotlin projects [here](native/cocoapods.html#add-dependencies-on-pod-libraries).
+Find examples in the [Kotlin with CocoaPods sample](https://github.com/Kotlin/kotlin-with-cocoapods-sample).
+
+### Updated integration with Xcode
+
+To work correctly with Xcode, Kotlin requires some Podfile changes:
+
+* If your Kotlin Pod has any Git, HTTP, or specRepo pod dependency, you should also specify it in the Podfile.
+* When you add a library from the custom spec, you also should specify the [location](https://guides.cocoapods.org/syntax/podfile.html#source)
+    of specs at the beginning of your Podfile.
+
+Now integration errors have a detailed description in IDEA. So if you have problems with your Podfile, you will immediately know how to fix them.
+
+Learn more about creating Kotlin pods [here](native/cocoapods.html#use-a-kotlin-gradle-project-as-a-cocoapods-dependency).
+
+[**Back to top**](#)
+
+## Standard library
+
+### Extensions for java.nio.file.Path
+
+> Extensions for `java.nio.file.Path` are [experimental](evolution/components-stability.html) in 1.4.20. They may change incompatibly in future versions.
+{:.note}
+
+Now the standard library provides experimental extensions for `java.nio.file.Path`.
+Working with the modern JVM file API in an idiomatic Kotlin way is now similar to working with `java.io.File` extensions
+from the `kotlin.io` package.
+
+The extensions are available in the `kotlin.io.path` package in the `kotlin-stdlib-jdk7` module.
+In order to use them, you need to opt-in to the experimental annotation `@ExperimentalPathApi`.
+
+```kotlin
+// construct path with the div (/) operator
+val baseDir = Path("/base")
+val subDir = baseDir / "subdirectory" 
+
+// list files in a directory
+val kotlinFiles: List<Path> = Path("/home/user").listDirectoryEntries("*.kt")
+```
+
+### Improved String.replace function performance
+
+The new implementation of `String.replace()` function speeds up the execution in certain cases.
```

Comment:
`certain` motivates to list these cases.

Reply:
@koshachy can you help with specific cases? 
If there are *not many* cases, we should mention when exactly the performance grows.
If this feature helps in *most cases*, then we can just replace `certain` with a more general phrase.

Reply:
Oh, maybe I said it in the wrong terms.
The next sentence is about the essential thing: now the function has different behavior for case-sensitive and case-insensitive strings.

I think "The new implementation of `String.replace()` speeds up the function execution." or smth like that